server/gui/utils.py

```diff
@@ -13,13 +13,49 @@ class WorkerSignals(QObject):
     Defines the signals available from a running worker thread.
     Supported signals are:
         finished
+        ready
         error - `str` error message
         result - `object` data returned from processing, anything
     """
     finished = Signal()
     error = Signal(str)
     result = Signal(object)
+    ready = Signal()
+
+
+class SiteReadySignals(QObject):
```

Comment:
Any reason this isn't a subclass of EmittingProcess (like Worker)?

Reply:
Because the signals aren't tied to the worker being a subprocess or a thread (If I understand your question correctly).